src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes/ApiControllerCodeFixer.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = "UA0013 CodeFix Provider")]
+    public class ApiControllerCodeFixer : CodeFixProvider
+    {
+        public const string GoodNamespace = "Microsoft.AspNetCore.Mvc";
+        public const string GoodClassName = "Controller";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ApiControllerAnalyzer.DiagnosticId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root is null)
+            {
+                return;
+            }
+
+            var node = root.FindNode(context.Span, false, true);
```

Comment:
i recommend named arguments for these constants.  i, for one, forget what they are without the names :)

Reply:
Actually, I only included these because I borrowed that from another analyzer. Unit test shows we don't need these so I'll drop them.